src/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -337,109 +332,183 @@ contract BLSRegistryCoordinatorWithIndices is EIP712, Initializable, IBLSRegistr
-        }
+        emit OperatorSocketUpdate(idToRegister, socket);
 
-        // set the operatorId to quorum bitmap history
-        _operatorBitmapHistory[operatorId].push(QuorumBitmapUpdate({
-            updateBlockNumber: uint32(block.number),
-            nextUpdateBlockNumber: 0,
-            quorumBitmap: uint192(quorumBitmap)
-        }));
-
-        // if the operator is not already registered, then they are registering for the first time
-        if (_operatorInfo[operator].status != OperatorStatus.REGISTERED) {
-            _operatorInfo[operator] = Operator({
-                operatorId: operatorId,
+        if (_operatorInfo[operatorToRegister].status != OperatorStatus.REGISTERED) {
+            _operatorInfo[operatorToRegister] = Operator({
+                operatorId: idToRegister,
                 status: OperatorStatus.REGISTERED
             });
 
-            emit OperatorRegistered(operator, operatorId);
+            emit OperatorRegistered(operatorToRegister, idToRegister);
+        }
+
+        /**
+         * Register the operator with the BLSPubkeyRegistry, StakeRegistry, and IndexRegistry
+         */
+        bytes32 registeredId = blsPubkeyRegistry.registerOperator(operatorToRegister, quorumNumbers);
+        require(registeredId == idToRegister, "BLSRegistryCoordinatorWithIndices._registerOperator: operatorId mismatch");
+        (uint96[] memory registeredStakes, uint96[] memory totalStakes) = 
+            stakeRegistry.registerOperator(operatorToRegister, idToRegister, quorumNumbers);
+        uint32[] memory numOperatorsPerQuorum = indexRegistry.registerOperator(idToRegister, quorumNumbers);
+
+        /**
+         * Now that we've registered the operator, validate that the new total operators
+         * in each quorum do not exceed the maximum. 
+         * 
+         * If they do and we're performing churn, we deregister the corresponding operator 
+         * specified in `operatorKickParams`
+         */
+
+        uint256 kickIndex = 0;
+        for (uint256 i = 0; i < quorumNumbers.length; i++) {
+            uint8 quorumNumber = uint8(quorumNumbers[i]);
+            
+            OperatorSetParam memory operatorSetParams = _quorumParams[quorumNumber];
+            bool exceedsOperatorCapacity = numOperatorsPerQuorum[i] > operatorSetParams.maxOperatorCount;
+
+            /**
+             * If we exceed the max operator count for this quorum, we need to replace an existing operator
+             * with a new one.
+             */
+            if (exceedsOperatorCapacity) {
+                require(performChurn, "BLSRegistryCoordinatorWithIndices._registerOperator: churn required for overfilled quorum");
+                
+                // Validate that `operatorToRegister` can replace the operator specified in the kick params
+                _validateChurn({
+                    quorumNumber: quorumNumber,
+                    totalQuorumStake: totalStakes[i],
```

Comment:
this is now getting validated against the _new_ total stakes, when before I think it was the _old_ total stakes (please correct me if I'm wrong!).
I don't think either option is "incorrect" but we should make the intention clear

Reply:
I think it's actually the same as before, glad you caught this though.

In both prev and cur implementation, we're updating the incoming operator's stake (and the quorum total stake) when the operator is registered.

We're then doing this validation against up-to-date info for the incoming operator and total, but potentially-old info the for operator getting churned.

Reply:
ahhh OK. Makes sense; I guess I hadn't considered the two options and it just occurred to me then.

Reply:
@gpsanant would appreciate your confirming that this is correct.